internal/saucecloud/cloud.go

```diff
@@ -867,12 +868,54 @@ func (r *CloudRunner) getAvailableVersionsMessage(frameworkName string) string {
 	return m
 }
 
-func (r *CloudRunner) getHistory(launchOrder config.LaunchOrder) (insights.JobHistory, error) {
+func (r *CloudRunner) getHistory(launchOrder config.LaunchOrder, sources []string) (insights.JobHistory, error) {
 	user, err := r.UserService.User(context.Background())
 	if err != nil {
 		return insights.JobHistory{}, err
 	}
-	return r.InsightsService.GetHistory(context.Background(), user, launchOrder)
+
+	var completeJobHistory []insights.JobHistory
+
+	for _, source := range sources {
+		jobHistory, err := r.InsightsService.GetHistory(context.Background(), user, launchOrder, source)
+		if err != nil {
+			return insights.JobHistory{}, err
+		}
+		completeJobHistory = append(completeJobHistory, jobHistory)
+	}
+	return mergeJobHistories(completeJobHistory), nil
+}
+
+func mergeJobHistories(histories []insights.JobHistory) insights.JobHistory {
+	testCasesMap := map[string]insights.TestCase{}
+	for _, history := range histories {
+		for _, tc := range history.TestCases {
+			addTestCase(&testCasesMap, tc)
+		}
+	}
+	var testCases []insights.TestCase
+	for _, tc := range testCasesMap {
+		testCases = append(testCases, tc)
+	}
+	sort.Slice(testCases, func(i, j int) bool {
+		return testCases[i].FailRate > testCases[j].FailRate
+	})
+	return insights.JobHistory{
+		TestCases: testCases,
+	}
+}
+
+// addTestCase adds the insights.TestCase in the map[string]insights.TestCase
+// If there is already one with the same name, only the highest fail rate is kept.
+func addTestCase(mp *map[string]insights.TestCase, tc insights.TestCase) {
```

Comment:
Doesn't have to be a pointer.

```suggestion
func addTestCase(mp map[string]insights.TestCase, tc insights.TestCase) {
```